packages/@aws-cdk/aws-sqs/lib/queue.ts

```diff
@@ -200,6 +200,7 @@ export class Queue extends QueueBase {
       public readonly encryptionMasterKey = attrs.keyArn
         ? kms.Key.fromKeyArn(this, 'Key', attrs.keyArn)
         : undefined;
+      public readonly fifo = attrs.queueArn.endsWith('.fifo') ? true : false;
```

Comment:
You must verify that queueArn is not a token if you parse it: `Token.isUnresolved(attrs.queueArn)`.

Reply:
I'm going to switch this to use `queueName` instead of `queueArn`. I see we do all of the checks and the right things to the ARN 10 lines prior to this one.